lib/HLSL/HLOperationLower.cpp

```diff
@@ -5416,11 +5419,14 @@ Value *GenerateCBLoadLegacy(Value *handle, Value *legacyIdx,
 
 Value *TranslateConstBufMatLdLegacy(Type *matType, Value *handle,
                                     Value *legacyIdx, bool colMajor, OP *OP,
-                                    const DataLayout &DL,
+                                    bool memElemRepr, const DataLayout &DL,
                                     IRBuilder<> &Builder) {
   unsigned col, row;
   Type *EltTy = HLMatrixLower::GetMatrixInfo(matType, col, row);
 
+  if (EltTy->isIntegerTy(1) && memElemRepr)
```

Comment:
This should rather be
`Type *EltTy = HLMatrixLower::LowerMatrixType(matType, /*forMem*/memElemRepr)->getVectorElementType();`
(changed locally)